lib/ckan_client.py

```diff
@@ -0,0 +1,63 @@
+from .util import ckan_auth_api
+from .util import ckan_upload_api
+from .file import FileSystem
+
+
+class Uploader:
+    '''
+    This is a class for authentiacation and uploading the
+    data to the cloud storage
+
+    Attributes:
+        base_url (str): url to the server
+        auth_token (str): authetication token for the request
+        organization_id (str): name of the organization
+        dataset_id (str): name of the dataset
+    '''
+
+    def __init__(self, base_url, auth_token, organization_id, dataset_id):
+        self.base_url = base_url
+        self.auth_token = auth_token
+        self.organization_id = organization_id
+        self.dataset_id = dataset_id
+
+    def ckan_authz(self):
+        # Get the JWT token from CkanAuthz
+
+        scope = 'obj:{}/{}/*:write'.format(self.organization_id, self.dataset_id)
+        response = ckan_auth_api.get_jwt_from_ckan_authz(self.base_url, self.auth_token, scope)
+        return response
+
+    def push(self, file, token, on_progress=None):
```

Comment:
Is there a reason you pass token here? It seems a simpler api for (users that) you have:

```
def push(self, file, on_progress)
```

And then this takes care of getting the token.

Also what is file? is it actually a file like object, a path on disk or ...